### PR TITLE
read+write session

### PR DESCRIPTION
**Pull request by andrewkmin**
## Summary & Motivation
Implements login + read/write sessions.

Also adds documentation/does a lil cleanup

## How I Tested These Changes
Add an example similar to Email Auth

## Did you add a changeset?

If updating one of our packages, you'll likely need to add a changeset to your PR. To do so, run [`pnpm changeset`](https://pnpm.io/using-changesets#adding-new-changesets). `pnpm changeset` will generate a file where you should write a human friendly message about the changes. Note how this ([example](https://github.com/tkhq/sdk/blob/b409cd06790f011bf939adcf0755499b8e7497ae/.changeset/extra-http-exports.md?plain=1#L1)) includes the package name (should be auto added by the command) along with the type of [semver change (major.minor.patch)](https://semver.org/) (which you should set).

These changes will be used at release time to determine what packages to publish and how to bump their version. For more context see [this comment](https://github.com/tkhq/sdk/pull/67#issuecomment-1568838440).

### CODE REVIEW

File: packages/sdk-browser/src/sdk-client.ts [hunk 1]
```diff
@@ -293,8 +382,8 @@ export class TurnkeyPasskeyClient extends TurnkeyBrowserClient {
         {
           apiKeyName: `Session Key ${String(Date.now())}`,
           publicKey,
-          expirationSeconds: expirationSeconds ?? DEFAULT_SESSION_EXPIRATION,
-          curveType: curveType ?? "API_KEY_CURVE_P256",
+          expirationSeconds,
+          curveType: "API_KEY_CURVE_P256",
```
Comment (andrewkmin) [inline packages/sdk-browser/src/sdk-client.ts:386]: By default all embedded API keys will be P256 keys

**Review by moe-dev — APPROVED** [review]
Awesome 🙏

File: packages/sdk-browser/src/sdk-client.ts [hunk 2]
```diff
@@ -207,9 +248,39 @@ export class TurnkeyBrowserClient extends TurnkeySDKClientBase {
...
+    });
+
+    // store auth bundle in local storage
+    await setStorageValue(StorageKeys.ReadWriteSession, {
+      authBundle: readWriteSessionResult!.credentialBundle,
+      sessionExpiry: Date.now() + Number(expirationSeconds) * 1000,
+    });
+
+    return readWriteSessionResult;
+  };
```
Comment (taylorjdawson) [inline packages/sdk-browser/src/sdk-client.ts:283]: @andrewkmin I assume I could use this method like this? 
```
authIframeClient.loginWithReadWriteSession(authIframeClient?.iframePublicKey)
```
if the authIframeClient already has this public key I wonder if this method could be on the `TurnkeyIframeClient` so it doesn't have to be passed in?

Comment (andrewkmin) [reply packages/sdk-browser/src/sdk-client.ts:283]: > authIframeClient.loginWithReadWriteSession(authIframeClient?.iframePublicKey)

Yeah that's exactly right!

> if the authIframeClient already has this public key I wonder if this method could be on the TurnkeyIframeClient so it doesn't have to be passed in?

Currently, any of the clients can initiate a `login with read write session` activity (e.g. `passkeyClient.loginWithReadWriteSession` is valid as well). You're right in that the TurnkeyIframeClient could override the base implementation (that lives on `TurnkeyBrowserClient`) and use something like `this.iframePublicKey()`

Comment (taylorjdawson) [reply packages/sdk-browser/src/sdk-client.ts:283]: tightt that would be so slicckkk!

**Pull request merged**